### PR TITLE
EFK stack using OpenEBS as the volume provisioner

### LINKED ISSUE #611: Running EFK stack on k8s using OpenEBS as the software defined storage
## Is this a BUG REPORT or FEATURE REQUEST?
 FEATURE REQUEST 

 Choose one: BUG REPORT or FEATURE REQUEST

**What happened**:
- Provide specifications (service, pvc, statefulset etc..) for running an EFK(Elasticsearch, FluentD, Kibana) cluster using OpenEBS for storage. 

**What you expected to happen**:
After the PR, 
kubectl apply -f <foldername with the specs> and it should create an EFK cluster on kubernetes with OpenEBS as the volume manager.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by Hrishike**
**What this PR does / why we need it**:
Cluster level log management using the EFK stack backed by OpenEBS. 
Refers #611 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #611 

**Special notes for your reviewer**:

### CODE REVIEW

File: k8s/openebs-operator.yaml [hunk 1]
```diff
@@ -74,7 +74,7 @@ spec:
         - name: DEFAULT_REPLICA_IMAGE
           value: "openebs/jiva:0.4.0"
         - name: DEFAULT_REPLICA_COUNT
-          value: "2"
+          value: "1"
```
Comment (kmova) [inline k8s/openebs-operator.yaml]: @Hrishike -- Can we set 2 itself as default.. 

An application that requires only one replica, set "replica: 1" in the volumes storage class. Example is openebs-standalone class in https://github.com/openebs/openebs/blob/master/k8s/openebs-storageclasses.yaml

Even though this depends on the resolution of : https://github.com/openebs/openebs/issues/645

Other than this.. the changes are good to be merged!

Comment (Hrishike) [reply k8s/openebs-operator.yaml]: @kmova sure. sounds apt. Done. Changed the default replicas to 2.

**Comment by Hrishike** [comment]
Updated with comments from @kmova . 
I also realized that this PR also contains initial commits for #663 and #667. 
Could you please review these? I could pull them back if required as well, so as to not get too many changes on at the same time in one PR.

**Comment by Hrishike** [comment]
@manigandham . It is still in the works. I am playing with the scenario where a Pod failover and rescheduling of the pod

**Comment by nitisuryawanshi** [comment]
@Hrishike Thank you so much for your PR contribution! We are announcing you as the winner of the best PR for hacktoberfest and would like to mention you in our tweets, thanking you for your contribution. Can you please send me your twitter handle on this mail ID <niti.suryawanshi@openebs.io> ?

**Pull request merged**